### ISSUE TITLE
[Tree] Disabled tree nodes are not announced as 'dimmed/unavailable'

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3624]
### 😯 Current Behavior

Disabled tree node use `disabled` attribute which is not properly announced by screen readers for that role of an element. Screen reader reads disabled node only as it name without any additional information. 

Disabled tree nodes have `tabindex='0'` which make them focusable so it may be even harder to distinguish which nodes are disabled and which are not.

### 🤔 Expected Behavior/possible solution?

Disabled nodes should still be focusable, but they should use `aria-disabled` instead of `disabled`. Then screen readers will announce disabled nodes as `dimmed/unavailable`.

### 🖥️ Steps to Reproduce

1. With screen reader turned on go to https://storybook.sanomalearning.design/?path=/story/navigation-tree-node--all
2. Place focus on disabled node
3. Verify if screen reader reads information 'dimmed' or 'unavailable' after the node's name

### Were you able to reproduce this in our storybook?

Yes

### If yes, provide the storybook URL

https://storybook.sanomalearning.design/?path=/story/navigation-tree-node--all

### ⚖️ Impact

This blocks our release or stops us from using this component

### Framework

Other

### What browsers are you seeing the problem on?

All

### Component version

0.3.0

### 👤 Your name

Kacper

### 🧢 Your product/team

SLDS

### 🎨 Your Theme(s)

Sanoma Learning
